### ISSUE TITLE
Rewrite posticle with Nom

### ISSUE THREAD

**Issue by barzamin** [rustodon/rustodon#110]
It's really nasty regex templating atm. We could fix this pretty easily by writing the posticle parser using Nom; i was originally going to and then didn't since it was easier to just port over the twitter-text regexes like [QuietMisdreavus's egg-mode](https://github.com/QuietMisdreavus/twitter-rs)

**Comment by measlytwerp** [comment]
I've just been exploring the parser landscape, tested nom, peg, pest and a couple of others. I'd be comfortable tackling this in either nom or pest, but the latter is my preference as it's just as fast and much easier for a new developer to use.

Is there a good writeup somewhere of what the syntax is?
Would it be ok to use pest instead of nom?

**Comment by barzamin** [comment]
if you're interested in doing something with Pest, go right ahead! i'd definitely be fine with that. i can write up a semi-formalized spec of what we want to parse out, too, just not _immediately_.

**Comment by measlytwerp** [comment]
Yeah, that'd be greatly appreciated, I'm fluent in regex though so I'll make a start on it. No hurry though, selling our house and have two open days scheduled this weekend so that may limit the time I'm available.
